controllers/mcad_controller.go

```diff
@@ -124,6 +124,8 @@ func (r *MCADReconciler) DeleteResource(params *MCADParams, template string, fns
 // +kubebuilder:rbac:groups=extensions,resources=replicasets,verbs=get;list;watch
 // +kubebuilder:rbac:groups=policy,resources=poddisruptionbudgets,verbs=get;list;watch
 // +kubebuilder:rbac:groups=storage.k8s.io,resources=csidrivers;csinodes;csistoragecapacities,verbs=get;list;watch
+// +kubebuilder:rbac:groups=authentication.k8s.io,resources=tokenreviews,verbs=create
```

Comment:
Actually, why do we need this?

Reply:
I'm not 100% sure, but this was brought up to me by @sutaakar as the e2e tests were failing when manually adding the tokenreviews, but when using kubebuilder to generate in the file instead, the e2e tests pass...

Reply:
I should clarify, why do we need tokenreviews at all?

Reply:
Previously I was getting this error: 
```
2023-07-26T09:16:08Z	INFO	Creating	{"name": "/mcad-mcad-controller-kube-scheduler-crb", "type": "rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding"}
2023-07-26T09:16:08Z	ERROR	Reconciler error	{"controller": "mcad", "controllerGroup": "codeflare.codeflare.dev", "controllerKind": "MCAD", "MCAD": {"name":"mcad","namespace":"codeflare-system"}, "namespace": "codeflare-system", "name": "mcad", "reconcileID": "4c657377-3718-4245-bd49-fa82bb5da2c6", 
"error": "clusterrolebindings.rbac.authorization.k8s.io \"mcad-mcad-controller-kube-scheduler-crb\" is forbidden: user \"system:serviceaccount:openshift-operators:codeflare-operator-controller-manager\" 
(groups=[\"system:serviceaccounts\" \"system:serviceaccounts:openshift-operators\" \"system:authenticated\"]) is attempting to grant RBAC permissions not currently held:\n{APIGroups:[\"authentication.k8s.io\"], Resources:[\"tokenreviews\"], Verbs:[\"create\"]}\n{APIGroups:[\"authorization.k8s.io\"], Resources:[\"subjectaccessreviews\"], Verbs:[\"create\"]}"} 
```
My understanding is that it seems that the Operator needed permission to create tokenreviews as it performs identity verification. In this case, the Operator receives requests from Prometheus when scraping metrics. [ServiceMonitor with bearerTokenFile](https://github.com/project-codeflare/codeflare-operator/blob/ca3baf682b6174e0a6424c1fae6734071237630a/config/prometheus/monitor.yaml#L12)

I suppose if we don't want to keep this, I could remove it? Please let me know what would be best to do